docs(suspensive.org): add docs for shouldCatch at `ErrorBoundary` (#1085)
suspensive/react: docs/suspensive.org/src/pages/docs/react/ErrorBoundary.en.mdx

</Sandpack>

### props.shouldCatch

shouldCatch determines whether `<ErrorBoundary/>` should catch errors based on conditions.

> It accepts three criteria: Boolean, Constructor, and Callback, and defaults to true.
> 
> <Tabs items={['Boolean', 'Constructor', 'Callback']}>
>   <Tabs.Tab>

Reviewer: ```suggestion
It accepts three criteria: Boolean, ErrorConstructor, and Callback, and defaults to true.

<Tabs items={['Boolean', 'ErrorConstructor', 'Callback']}>
  <Tabs.Tab>
```

> ```tsx {10}

Reviewer: I think {10} is good, but /shouldCatch/ also could be same benefit and good to maintain 

```suggestion
```tsx /shouldCatch/
```

Author: oh, I think it's definitely better. 👍

import { ErrorBoundary } from '@suspensive/react'
import { createElement } from 'react'
import { CustomError } from './CustomError'
import { CustomErrorAfter2s } from './CustomErrorAfter2s'

export const Example = () => {
  return (
    <ErrorBoundary fallback={({ error }) => <>{error.message} of Parent</>}>
      <ErrorBoundary
        shouldCatch={false}
        fallback={({ error }) => <>{error.message} of Child</>}
      >
        <CustomErrorAfter2s />
      </ErrorBoundary>
    </ErrorBoundary>
  )
}
```
  </Tabs.Tab>
  <Tabs.Tab>

> ```tsx {10}

Reviewer: ```suggestion
```tsx /shouldCatch/
```

  )
}
```

<Sandpack>

```tsx Example.tsx active
import { ErrorBoundary } from '@suspensive/react'
import { createElement } from 'react'
import { CustomError } from './CustomError'
import { CustomErrorAfter2s } from './CustomErrorAfter2s'

export const Example = () => {
  return (
    <ErrorBoundary fallback={({ error }) => <>{error.message} of Parent</>}>
      <ErrorBoundary
        shouldCatch={CustomError}
        fallback={({ error }) => <>{error.message} of Child</>}
      >
        <CustomErrorAfter2s />
      </ErrorBoundary>
    </ErrorBoundary>
  )
}
```

```tsx CustomError.ts

export class CustomError extends Error {
  constructor(...args: ConstructorParameters<ErrorConstructor>) {
    super(...args)
    console.error(...args)
  }
}
```

```tsx CustomErrorAfter2s.tsx
import { useState, useEffect } from "react";
import { CustomError } from "./CustomError";

export const CustomErrorAfter2s = () => {
  const [asyncState, setAsyncState] = useState<
    { isError: true; error: CustomError } | { isError: false; error: null }
  >({
    isError: false,
    error: null
  });

  useEffect(() => {
    setTimeout(() => {
      setAsyncState({
        isError: true,

>         error: () => new CustomError("unauthorized")

Reviewer: Could you remove all unnecessary context in this doc please? In my opinion, text "unauthorized" could make misunderstanding concept shouldCatch

```suggestion
        error: () => new CustomError("⚠️ Error made by CustomError Constructor")
```

Author: I agree, I changed it!

> ```tsx Example.tsx active
> import { ErrorBoundary } from '@suspensive/react'
> import { createElement } from 'react'
> import { CustomError } from './CustomError'
> import { CustomErrorAfter2s } from './CustomErrorAfter2s'
> 
> export const Example = () => {
>   return (
>     <ErrorBoundary fallback={({ error }) => <>{error.message} of Parent</>}>
>       <ErrorBoundary
>         shouldCatch={CustomError}
>         fallback={({ error }) => <>{error.message} of Child</>}
>       >
>         <CustomErrorAfter2s />
>       </ErrorBoundary>
>     </ErrorBoundary>
>   )
> }
> ```
> 
> ```tsx CustomError.ts
> 
> export class CustomError extends Error {
>   constructor(...args: ConstructorParameters<ErrorConstructor>) {
>     super(...args)
>     console.error(...args)
>   }
> }
> ```
> 
> ```tsx CustomErrorAfter2s.tsx
> import { useState, useEffect } from "react";
> import { CustomError } from "./CustomError";
> 
> export const CustomErrorAfter2s = () => {
>   const [asyncState, setAsyncState] = useState<
>     { isError: true; error: CustomError } | { isError: false; error: null }
>   >({
>     isError: false,
>     error: null
>   });
> 
>   useEffect(() => {
>     setTimeout(() => {
>       setAsyncState({
>         isError: true,
>         error: () => new CustomError("unauthorized")
>       });
>     }, 2000);
>   }, []);
> 
>   if (asyncState.isError) {
>     throw asyncState.error();
>   }
> 
>   return <>No error</>;
> };
> ```
> 

Reviewer: In my opinion, no depth and one file could make doc easier to understand .

```suggestion
```tsx Example.tsx active
import { useState, useEffect } from "react";
import { ErrorBoundary } from '@suspensive/react'
import { createElement } from 'react'

export const Example = () => {
  return (
    <ErrorBoundary fallback={({ error }) => <>{error.message} of Parent</>}>
      <ErrorBoundary
        shouldCatch={CustomError}
        fallback={({ error }) => <>{error.message} of Child</>}
      >
        <CustomErrorAfter2s />
      </ErrorBoundary>
    </ErrorBoundary>
  )
}

class CustomError extends Error {
  constructor(...args: ConstructorParameters<ErrorConstructor>) {
    super(...args)
    console.error(...args)
  }
}

export const CustomErrorAfter2s = () => {
  const [asyncState, setAsyncState] = useState<
    { isError: true; error: CustomError } | { isError: false; error: null }
  >({
    isError: false,
    error: null
  });

  useEffect(() => {
    setTimeout(() => {
      setAsyncState({
        isError: true,
        error: () => new CustomError("unauthorized")
      });
    }, 2000);
  }, []);

  if (asyncState.isError) {
    throw asyncState.error();
  }

  return <>No error</>;
};
```

```

Author: Oh, Okay. I fixed it. 🧐

</Sandpack>

<br />

<Sandpack>

```tsx Example.tsx active
import { ErrorBoundary } from '@suspensive/react'
import { createElement } from 'react'
import { CustomError } from './CustomError'
import { ErrorAfter2s } from './ErrorAfter2s'

export const Example = () => {
  return (
    <ErrorBoundary fallback={({ error }) => <>{error.message} of Parent</>}>
      <ErrorBoundary
        shouldCatch={CustomError}
        fallback={({ error }) => <>{error.message} of Child</>}
      >
        <ErrorAfter2s />
      </ErrorBoundary>
    </ErrorBoundary>
  )
}
```

```tsx CustomError.ts

export class CustomError extends Error {
  constructor(...args: ConstructorParameters<ErrorConstructor>) {
    super(...args)
    console.error(...args)
  }
}
```

```tsx /ErrorAfter2s.tsx
import { useState, useEffect } from 'react'

export const ErrorAfter2s = () => {
  const [asyncState, setAsyncState] = useState<{ isError: true; error: Error } | { isError: false; error: null }>({
    isError: false,
    error: null,
  })

  useEffect(() => {
    setTimeout(() => {

>       setAsyncState({ isError: true, error: { status: 401, message: 'unauthorized' } })

Reviewer: Same here
